### ISSUE TITLE
Prompt in `read -p promt` interpreted as variable name

### ISSUE THREAD

**Issue by schaetzc** [koalaman/shellcheck#1599]
#### Bug (false negative)
- Rule Id: SC2154, SC2034
- My shellcheck version: ``shellcheck --version` says 0.6.0, `pacman -Qi` says 0.6.0-120
- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
The following script will prompt `something` and read input to the default variable `REPLY`. Afterwards the unassigned variable `$something` is accessed.
```sh
#! /bin/bash
read -r -p something
echo "$something"
```
#### Here's what shellcheck currently says:
There is no output. Shellcheck doesn't find a problem even though there is one (false negative).

#### Here's what I wanted or expected to see:
```
echo "$something"
          ^-- SC2154: something is referenced but not assigned.
```
## More Information

Shellcheck seems to misinterpret the prompt argument given to `read`'s `-p` option as a variable name. I did some further experiments:

For the script
```sh
#! /bin/bash
read -r -p something
echo "$REPLY"
```
shellcheck prints a warning even though there is no problem (false positive).
```
read -r -p something
           ^-------^ SC2034: something appears unused. Verify use (or export if used externally).
```
I expected shellcheck to remain silent.

For the script
```sh
#! /bin/bash
read -r -p a b
echo "$a"
echo "$b"
```
shellcheck remains silent even though there is a problem (false negative). I expected the output
```
echo "$a"
      ^-- SC2154: a is referenced but not assigned.
```

**Comment by schaetzc** [comment]
Some further experiments showed that also other non-variable parameters are interpreted as variable names. This includes (but is not necessarily limited to) `read -d a`, `read -i a` and `read -u a` – in each of this cases shellcheck interprets `a` as a variable name even though it is a delimiter, initial text, or file descriptor.

Note that `a` has to be a valid variable identifier for shellcheck to give the false positive SC2034 (appears unused) and the false negative, that is, missing warning SC2154 (referenced but not assigned).

**Comment by svanzoest** [comment]
I am getting a similar error on the following
```bash
  while read -a files -r
  do
    if ! downloadFile "${files[0]}" "${files[1]}"; 
    then
      ((++downloadFailures))
      echo "downloadFailures ${downloadFailures}"
    fi

  done < "./filelist.txt"
```
`  ^-- SC2154: files is referenced but not assigned.`
